### DISCUSSION TITLE
await loading stores

### DISCUSSION THREAD

**Discussion by islamzaoui** [sveltejs/kit#10722] [Q&A]
Hello

When I subscribe to a store with `$<storename>` for a couple of seconds will be undefined and then load its data.

Is there any way to wait for loading stores?

**Comment by stephane-vanraes** [comment]
you can use `{#await}` if you need it in the markup, but it sounds like you are working with some custom store, it is probably better to at least add a default value to your store

alternatively make a custom store with a loading, error, data state and use `if` statements to differ between the states (a quick search should give you example of this pattern)
